packages/vscode-extension/src/extension.ts

```diff
@@ -273,31 +321,24 @@ async function findAppRootFolder() {
     return appRoot;
   }
 
-  const metroConfigUri = await findSingleFileInWorkspace(
-    "**/metro.config.{js,ts}",
-    "**/node_modules"
-  );
-  if (metroConfigUri) {
-    return Uri.joinPath(metroConfigUri, "..").fsPath;
-  }
-
-  const appConfigUri = await findSingleFileInWorkspace("**/app.config.{js,ts}", "**/node_modules");
-  if (appConfigUri) {
-    return Uri.joinPath(appConfigUri, "..").fsPath;
-  }
+  const appRootCandidates = await findAppRootCandidates();
 
-  const rnPackageLocation = await findSingleFileInWorkspace(
-    "**/node_modules/react-native/package.json",
-    null
-  );
-  if (rnPackageLocation) {
-    return Uri.joinPath(rnPackageLocation, "../../..").fsPath;
+  if (appRootCandidates.length > 1) {
+    const openLaunchConfigButton = "Open Launch Configuration";
+    window
+      .showErrorMessage(
+        `multiple reat-native applications were detected in the workspace, ${appRootCandidates[0]} were chosen, but you can change the root of your application in lunch configuration.`,
```

Comment:
add some text that says that in order to get rid of this error, people should specify app root in the launch configuration

Reply:
Also, "lunch configuration" typo. 🥪